src/deploymentTemplate.ts

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as utilities from './utilities';
+import { Constants } from './constants';
+import { TelemetryClient } from './telemetryClient';
+
+var request = require('request');
+
+export class DeploymentTemplate {
+    constructor() {}
+
+    public displayDeploymentTemplateMenu() {
+
+        TelemetryClient.sendEvent('deployment');
+        // currently only one option is available, so first menu won't be displayed yet
+        this.selectQuickstartTemplate();
+    }
+
+    public selectQuickstartTemplate() {
+
+        // get list of directories from here:
+        // https://api.github.com/[repo_user/repo_name]/contents/
+        let repo: string = utilities.getCodeConfiguration<string>(null, Constants.Config_deploymentTemplatesGitHubRepo);
+
+        var http = require('https');
+        let __this = this;
+
+            http.get({
+                host: 'api.github.com',
+                path: '/repos/' + repo + '/contents/',
+                headers: { 'User-Agent': 'VSC Ansible Extension'}
+            }, function(response) {
+                // Continuously update stream with data
+                var body = '';
+                response.on('data', function(d) {
+                    body += d;
+                });
+                response.on('end', function() {
+        
+                    // Data reception is done, do whatever with it!
+                    var parsed = JSON.parse(body);
+                    let items: vscode.QuickPickItem[] = [];
+            
+                    for (var i in parsed)
+                    {
+                        // list only directories and skip known directories that don't contain templates
+                        if (parsed[i].type == "dir" && parsed[i].name != "1-CONTRIBUTION-GUIDE" && parsed[i].name != ".github") {
```

Comment:
this logic seems only applies to azure-template repo structure. add strict condition check on repo name?

Reply:
i think perhaps we shouldn't check for that "1-CONTRIBUTION-GUIDE", just if user selects it we should display "No azuredeplot.json found"

Reply:
it's hard to condition check on users' repo structure. so suggesting to let user config template file full path directly instead of repo root path. Seems customized github repo not ready yet, can it be added later?